pkg/pillar/agentbase/apptracker.go

```diff
@@ -0,0 +1,556 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package agentbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// TriState - the boolean type with 3 states for generic AppInfoItems
+type TriState int
+
+const (
+	// NotSet indicates the initial value, which is 0.
+	NotSet TriState = iota
+	// False indicates the false value.
+	False
+	// True indicates the true value.
+	True
+)
+
+// String - convert the TriState to string
+func (t TriState) String() string {
+	return [...]string{"NotSet", "False", "True"}[t]
+}
+
+// MarshalJSON - convert the TriState to JSON
+func (t TriState) MarshalJSON() ([]byte, error) {
+	return json.Marshal(t.String())
+}
+
+func boolToTriState(b bool) TriState {
+	if b {
+		return True
+	}
+	return False
```

Comment:
Using True and False is likely to confuse the reader to think this is a boolean type. (types.Tristate uses unique values TS_ENABLED and TS_DISABLED for the non-zero values for that reason.)

Reply:
Yes, updated.